src/kws.c

```diff
@@ -210,25 +211,50 @@ static int ws_client_handshake(kws_t *kws)
 {
 	unsigned char nonce[16] = { 0 };
 	unsigned char enonce[128] = { 0 };
-	char req[2048] = { 0 };
+	char req[12048] = { 0 };
```

Comment:
Since the headers are dynamic, and we're now exceeding 2kb on the stack, I recommend switching this away from stack allocation and using heap allocation for req by using ks_mprintf further down instead of ks_snprintf and freeing after it's written out. This prevents large stack allocations, and also ensures large headers won't be truncated leaving the request in an undefined and unusable state.